_posts/2020-04-07-trying-out-julia.md

```diff
@@ -0,0 +1,227 @@
+---
+layout: post
+title: Trying out Julia programming language
+author: hnybom
+excerpt: >
+  Julia is a programming language with it's roots in the mathematics. The language itself is increasingly gaining momentum and that combined with the boom were seeing in data science demand makes it a very interesting subject. This post is about my journey starting from zero knowledge to, well, some knowledge of the language. So if you, like me, are interested in trying out Julia language keep on reading. If you are already familiar with the subject you can safely skip this one.
+tags:
+ - Julia
+---
+
+## Let's try out Julia
+
+As forewords, I have never tried Julia programming language before but I've read couple of headlines about it and they have peaked my interest on the subject. The main motivations to get into the language for me are
+
+* Julia seems to be gaining some momentum in both the programming and especially data science community.
+* Even as Julia is dynamically typed it has some static typing going on by enabling the programmer to declare a type if it makes the code more easy to understand. This is akin of Python's type annotation though it seems a little but more powerful. There is a long documentation about this at [docs.julialang.org/en/v1/manual/types/](https://docs.julialang.org/en/v1/manual/types/). One very definite upside I've learned is that you can only sub type abstract types and extending a concrete type is not possible, as a Spring framework user this warms my mind a lot ;) !
+* The benchmarks seems to indicate it's much faster than the main competition Python. This is due the fact the language is compiled to native code using LLVM.
+* You can utilize Python packages.
+* I have some background with MatLab and similar plotting stuff seems to be the forte of Julia.
+
+This post is mostly about plotting as it was the main motivation for me to get into Julia language. I'll try to do same kind of stuff I did with MatLab with Julia. Basically this means scripting some plotting and I won't delve into the finer points of e.g. building server side software.
```

Comment:
There should be a comma after Basically.

Also, Basically is a bit condescending filler word, so have to be a bit cautious on overusing it at least. You would say basically when you are of superior intellect, and mansplaining something to someone who does not understand. :)